tests/hpo/algorithms/test_nelder_mead_algorithm.py

```diff
@@ -0,0 +1,224 @@
+        nm.put_value(np.zeros(2), expected_value2)
+        try:
+            next(nm._wait_for_results(2))
+            raise AssertionError()
+        except UnexpectedVerticesUpdateError:
+            assert True
+
+
+def test_collect_enqueued_results_empty(nm: NelderMeadAlgorism) -> None:
+    # queue is Empty
+    result_vertices, result_values = nm._collect_enqueued_results()
+    assert result_vertices == []
+    assert result_values == []
+
+
+def test_collect_enqueued_results(nm: NelderMeadAlgorism) -> None:
+    # queue is not Empty
+    expected_vertex1 = np.array([1.0, 2.0])
+    expected_value1 = 1.0
+    nm.put_value(expected_vertex1, expected_value1, enqueue=True)
+    expected_vertex2 = np.array([3.0, 4.0])
+    expected_value2 = 2.0
+    nm.put_value(expected_vertex2, expected_value2, enqueue=True)
+
+    result_vertices, result_values = nm._collect_enqueued_results()
+    assert np.array_equal(result_vertices, [expected_vertex1, expected_vertex2])
+    assert result_values == [expected_value1, expected_value2]
+
+
+def test_initialize(search_space: dict[str, tuple[float, float]], nm: NelderMeadAlgorism) -> None:
+    for _ in range(len(search_space) + 1):
+        random_xs = nm.get_vertex()
+        for random_x, distribution in zip(random_xs, search_space.values(), strict=False):
+            assert isinstance(random_x, float) and distribution[0] <= random_x <= distribution[1]
+    try:
+        nm.get_vertex()
+        raise AssertionError()
+    except NelderMeadEmptyError:
+        assert True
+
+
+def test_initialize_enqueued1(search_space: dict[str, tuple[float, float]], nm: NelderMeadAlgorism) -> None:
+    # enqueued
+    enqueued_vertex = np.array([1.0, 2.0])
+    enqueued_value = 1.0
+    nm.put_value(enqueued_vertex, enqueued_value, enqueue=True)
+
+    for _ in range(len(search_space)):
+        random_xs = nm.get_vertex()
+        for random_x, distribution in zip(random_xs, search_space.values(), strict=False):
+            assert isinstance(random_x, float) and distribution[0] <= random_x <= distribution[1]
+
+    assert np.array_equal(nm.vertices, [enqueued_vertex])
+    assert nm.values == [enqueued_value]
+
+    try:
+        nm.get_vertex()
+        raise AssertionError()
+    except NelderMeadEmptyError:
+        assert True
+
+
+def test_initialize_enqueued2(nm: NelderMeadAlgorism) -> None:
+    # enqueued
+    enqueud_vertices = [np.array([1.0, 2.0]), np.array([2.0, 3.0]), np.array([4.0, 5.0]), np.array([6.0, 7.0])]
+    enqueud_values = [1.0, 4.0, 2.0, 3.0]
+    expected_vertex = np.array([-1.0, 0.0])
+
+    for enqueued_vertex, enqueued_value in zip(enqueud_vertices, enqueud_values, strict=False):
+        nm.put_value(enqueued_vertex, enqueued_value, enqueue=True)
+
+    x = nm.get_vertex()
+    assert np.array_equal(x, expected_vertex)
+
+
+@pytest.mark.parametrize(
+    "expected_results",  # (vertex, value, enqueue)
+    [
+        # reflect
+        ([([-1.0, 0.0], None, False)]),
```

Comment:
```suggestion
        [{"vertex": [-1.0, 0.0], "value": None, "enqueue": False}],
```
と書いた方が分かりやすいかも…？ お任せします．